### PR TITLE
Trying to fix the overall performance of the hub search

### PR DESCRIPTION
**Pull request by VamshiReddy02**
fixes: #1097 

Hey @itowlson, can you review this PR? I took up your suggestion from my last PR i.e., #1111. Please let me know if any further changes need to be made.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [ ] The `title, `template`, and `date` are all set
- [ ] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [ ] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [ ] Headings are using Title Case
- [ ] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors
- [ ] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

**Comment by itowlson** [comment]
@VamshiReddy02 I'm not qualified to review the detailed behaviour, but it's definitely the kind of thing I was thinking of!  Thanks!  @karthik2804 is much more familiar with this side of things and can hopefully take a look.

File: spin-up-hub/src/store.js [hunk 1]
```diff
@@ -115,22 +115,25 @@ const store = createStore({
       })
     },
     async loadSearchData(context) {
-      let data = await fetch(import.meta.env.VITE_API_HOST + "/static/hub-index-data.json")
-      let documents = await data.json()
+      let data = await fetch(import.meta.env.VITE_API_HOST + "/static/hub-index-data.json");
```
Comment (karthik2804) [inline spin-up-hub/src/store.js:118]: nit: I would not add semicolons here just to be consistent with the rest of the code.

**Review by karthik2804 — CHANGES_REQUESTED** [review]
Thanks for this @VamshiReddy02. This is a great starting place to address the search issue.

 I did some investigating into the query generation side of things. Based on the investigation, a couple of extra things could be tweaked in addition to this PR to improve the performance.
 
1. Remove the fuzzy matching in the query or at least limit it to 1 character [here](https://github.com/fermyon/developer/blob/cafc63a0aca48029fcf58ed9e7b1e4458cdaad7c/spin-up-hub/src/components/ContentListing.vue#L27)

1. Increase the threshold while filtering the matches to something higher than the current `0.5` which is really low [here](https://github.com/fermyon/developer/blob/cafc63a0aca48029fcf58ed9e7b1e4458cdaad7c/spin-up-hub/src/components/ContentListing.vue#L33)

Would you be open to adding those changes to this PR?

**Comment by VamshiReddy02** [comment]
Hey @karthik2804, thank you for the feedback. I made some changes which were requested. I'm still doubtful about the fuzzy matching and Increase in the threshold changes, can you check and let me know if there are any changes to be made?

File: spin-up-hub/src/components/ContentListing.vue [hunk 2]
```diff
@@ -24,13 +24,13 @@ export default {
             if (this.searchTerm) {
                 let updatedQuery = this.searchTerm
                     .split(" ")
-                    .map(word => word + '^2 ' + word + '* ' + word + '~2')
+                    .map(word => word + '^2 ' + word + '* ')
```
Comment (karthik2804) [inline spin-up-hub/src/components/ContentListing.vue:27]:
```suggestion
                    .map(word => word + '^2 ' + word + '*')
```

File: spin-up-hub/src/components/ContentListing.vue [hunk 3]
```diff
@@ -24,13 +24,13 @@ export default {
...
                     .join(' ');
                 let result = this.searchIndex.search(updatedQuery)
                 //get only confident results
                 let matches = []
                 result.map(k => {
-                    if (k.score < 0.5) {
+                    if (k.score < 1) {
```
Comment (karthik2804) [inline spin-up-hub/src/components/ContentListing.vue:33]: This is the right place but I would go higher with all the boosts we have, in my experiments something around 5 was good.

**Review by karthik2804 — COMMENTED** [review]
Just a couple of small tweaks but looks great overall! Thanks once again for handling this!

**Comment by VamshiReddy02** [comment]
@karthik2804 please review this PR.

File: spin-up-hub/src/store.js [hunk 4]
```diff
@@ -118,15 +118,15 @@ const store = createStore({
...
-        this.field('title')
+        this.field('title', { boost: 100 })
         this.field('content')
         this.field('language', { boost: 10 })
-        this.field('author'),
-          this.field('tags', { boost: 100 })
-        this.field('keywords', { boost: 100 })
+        this.field('author', { boost: 10 }),
+        this.field('tags', { boost: 10 })
+        this.field('keywords', { boost: 10 })
```
Comment (karthik2804) [inline spin-up-hub/src/store.js:126]: I think we should leave these fields as is, we want to match tags and keywords - keywords are a way to make sure we can specify alternative terms from tags. If someone wants to search for a particular author, they can search something along the lines of `author:<name>`

Comment (VamshiReddy02) [reply spin-up-hub/src/store.js]: Got it

**Review by karthik2804 — APPROVED** [review]

**Comment by karthik2804** [comment]
Thanks for this - this is great! I will merge once CI is done

**Pull request merged**